src/memray/_memray/record_reader.cpp

```diff
@@ -108,6 +108,35 @@ RecordReader::readVarint(size_t* val)
     }
 }
 
+bool
+RecordReader::readSignedVarint(ssize_t* val)
+{
+    size_t zigzag_val;
+    if (!readVarint(&zigzag_val)) {
+        return false;
+    }
+
+    *val = static_cast<ssize_t>((zigzag_val >> 1) ^ (~(zigzag_val & 1) + 1));
+    return true;
+}
+
+bool
+RecordReader::readStringDelta(std::string* cache, std::string* new_val)
```

Comment:
`cache` is a bit weird. What about `last_string` or similar? Technically is a cache of 1 element but if you don't know what's going on in general it can be a bit confusing